File: app/src/main/java/com/github/benji377/timety/ui/screens/StatsScreen.kt
package com.github.benji377.timety.ui.screens

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.LocalFireDepartment
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.Icon
import androidx.compose.material3.LinearProgressIndicator
import androidx.compose.material3.ListItem
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Text
import androidx.compose.material3.TopAppBar
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import com.github.benji377.timety.ui.components.RadialGraph
import com.github.benji377.timety.ui.components.XPBar
import com.github.benji377.timety.viewmodel.StatsViewModel

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun StatsScreen(viewModel: StatsViewModel) {
    val user by viewModel.user.collectAsState()
    val sessions by viewModel.allSessions.collectAsState()
    val todayFocusTime by viewModel.todayFocusTime.collectAsState()
    val categories by viewModel.allCategories.collectAsState()
    val distribution by viewModel.categoryDistribution.collectAsState()
    val insights by viewModel.insights.collectAsState()
    val weeklyData by viewModel.weeklyFocusData.collectAsState()

    Scaffold(
        topBar = { TopAppBar(title = { Text("Stats & Profile") }) }
    ) { padding ->
        LazyColumn(
            modifier = Modifier
                .padding(padding)
                .fillMaxSize()
                .padding(16.dp),
            verticalArrangement = Arrangement.spacedBy(16.dp)
        ) {
            // Profile Header
            item {
                Card(modifier = Modifier.fillMaxWidth()) {
                    Column(modifier = Modifier.padding(16.dp)) {
                        Row(verticalAlignment = Alignment.CenterVertically) {
                            Column(modifier = Modifier.weight(1f)) {
                                Text(
                                    text = user?.name ?: "Hero",
                                    style = MaterialTheme.typography.headlineSmall
                                )
                                // Dynamic title based on streak
                                val title = when {
                                    user?.currentStreak ?: 0 >= 30 -> "🔥 Focus Master"
                                    user?.currentStreak ?: 0 >= 14 -> "⚡ Unstoppable"
                                    user?.currentStreak ?: 0 >= 7 -> "🌟 On Fire"
                                    user?.currentStreak ?: 0 >= 3 -> "💪 Building Momentum"
                                    (user?.currentStreak ?: 0) > 0 -> "🎯 Focused"
                                    else -> "🚀 Ready to Focus"
                                }
                                Text(
                                    text = title,
                                    style = MaterialTheme.typography.bodyMedium,
                                    color = MaterialTheme.colorScheme.secondary
                                )
                            }
                            Icon(
                                imageVector = Icons.Default.LocalFireDepartment,
                                contentDescription = "Streak",
                                tint = MaterialTheme.colorScheme.error,
                                modifier = Modifier.size(32.dp)
                            )
                            Text(
                                text = "${user?.currentStreak ?: 0}",
                                style = MaterialTheme.typography.titleLarge
                            )
                        }

                        Spacer(modifier = Modifier.height(16.dp))

                        val currentXp = user?.xp ?: 0
                        val level = user?.level ?: 1
                        level * 100
                        val currentLevelXp = (level - 1) * 100
                        val xpInLevel = currentXp - currentLevelXp
                        val xpTarget = 100
                        val progress = (xpInLevel.toFloat() / xpTarget.toFloat()).coerceIn(0f, 1f)

                        XPBar(
                            progress = progress,
                            currentXp = xpInLevel,
                            targetXp = xpTarget,
                            level = level
                        )
                    }
                }
            }

            item {
                Text(text = "Today's Progress", style = MaterialTheme.typography.titleLarge)

                val target = user?.dailyFocusTarget ?: 1L
                val progress = (todayFocusTime.toFloat() / target.toFloat()).coerceIn(0f, 1f)
                val hours = todayFocusTime / 3600000
                val minutes = (todayFocusTime % 3600000) / 60000

                Box(modifier = Modifier.fillMaxWidth(), contentAlignment = Alignment.Center) {
                    RadialGraph(
                        progress = progress,
                        text = "${hours}h ${minutes}m",
                        modifier = Modifier.size(150.dp)
                    )
                }
            }

            // Weekly Chart
            item {
                Card(modifier = Modifier.fillMaxWidth()) {
                    Column(modifier = Modifier.padding(16.dp)) {
                        Text("This Week's Focus", style = MaterialTheme.typography.titleMedium)
                        Spacer(modifier = Modifier.height(12.dp))

                        weeklyData.forEach { (day, duration) ->
                            Column(modifier = Modifier
                                .fillMaxWidth()
                                .padding(vertical = 8.dp)) {
                                Row(
                                    modifier = Modifier.fillMaxWidth(),
                                    horizontalArrangement = Arrangement.SpaceBetween,
                                    verticalAlignment = Alignment.CenterVertically
                                ) {
                                    Text(day, style = MaterialTheme.typography.labelMedium)
                                    Text(
                                        "${duration / 60000} min",
                                        style = MaterialTheme.typography.labelSmall
                                    )
                                }
                                val maxDuration = weeklyData.values.maxOrNull() ?: 1L
                                val barProgress =
                                    if (maxDuration > 0) duration.toFloat() / maxDuration.toFloat() else 0f
                                LinearProgressIndicator(
                                    progress = { barProgress },
                                    modifier = Modifier
                                        .fillMaxWidth()
                                        .height(8.dp),
                                    color = MaterialTheme.colorScheme.primary
                                )
                            }
                        }
                    }
                }
            }

            // Insights
            if (insights.isNotEmpty()) {
                item {
                    Text(text = "Insights", style = MaterialTheme.typography.titleLarge)
                }

                items(insights) { insight ->
                    Card(
                        modifier = Modifier.fillMaxWidth(),
                        colors = CardDefaults.cardColors(
                            containerColor = MaterialTheme.colorScheme.secondaryContainer
                        )
                    ) {
                        Text(
                            text = insight,
                            modifier = Modifier.padding(12.dp),
                            style = MaterialTheme.typography.bodySmall
                        )
                    }
                }
            }

            if (distribution.isNotEmpty()) {
                item {
                    Text(
                        text = "Category Distribution",
                        style = MaterialTheme.typography.titleLarge
                    )
                }

                items(distribution.size) { index ->
                    val (catId, duration) = distribution.toList()[index]
                    val category = categories.find { it.id == catId }
                    val catHours = duration / 3600000
                    val catMinutes = (duration % 3600000) / 60000

                    Row(
                        modifier = Modifier
                            .fillMaxWidth()
                            .padding(vertical = 4.dp),
                        horizontalArrangement = Arrangement.SpaceBetween
                    ) {
                        Text(text = category?.name ?: "Unknown")
                        Text(text = "${catHours}h ${catMinutes}m")
                    }
                    LinearProgressIndicator(
                        progress = {
                            duration.toFloat() / sessions.sumOf { it.duration }.toFloat()
                        },
                        modifier = Modifier.fillMaxWidth(),
                        color = category?.let { Color(android.graphics.Color.parseColor(it.colorHex)) }
                            ?: MaterialTheme.colorScheme.primary
                    )
                }
            }

            item {
                Text(text = "Recent Sessions", style = MaterialTheme.typography.titleLarge)
            }

            items(sessions) { session ->
                ListItem(
                    headlineContent = { Text("Focus Session") },
                    supportingContent = { Text("${session.duration / 60000} minutes") },
                    trailingContent = { Text(session.rating?.name ?: "") }
                )
            }
        }
    }
}

File: lib/screens/stats_screen.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import '../providers/stats_provider.dart';
import '../providers/user_provider.dart';
import '../data/category.dart';
import '../widgets/radial_graph.dart';
import '../widgets/xp_bar.dart';
import '../data/user.dart';
import '../theme/app_theme.dart';
import 'daily_stats_screen.dart';

class StatsScreen extends StatelessWidget {
  const StatsScreen({super.key});

  @override
  Widget build(BuildContext context) {
    final statsProvider = context.watch<StatsProvider>();
    final userProvider = context.watch<UserProvider>();
    final user = userProvider.user;
    final theme = Theme.of(context);

    final todayFocusTime = statsProvider
        .getSessionsForDay(DateTime.now())
        .fold(0, (sum, s) => sum + s.duration);
    final insights = statsProvider.getInsights();
    final weeklyData = statsProvider.getWeeklyFocusData();
    final distribution = statsProvider.getCategoryDistribution();

    return Scaffold(
      body: SafeArea(
        child: ListView(
          padding: const EdgeInsets.all(16),
          children: [
            Text('Profile & Stats', style: theme.textTheme.headlineSmall),
            const SizedBox(height: 24),
            _buildProfileHeader(context, user),
            const SizedBox(height: 24),
            _buildTodayProgress(context, user, todayFocusTime),
            const SizedBox(height: 24),
            _buildWeeklyChart(context, weeklyData),
            const SizedBox(height: 24),
            if (insights.isNotEmpty) ...[
              Text('Insights', style: Theme.of(context).textTheme.titleLarge),
              const SizedBox(height: 12),
              ...insights.map(
                (insight) => Card(
                  color: theme.colorScheme.secondaryContainer,
                  margin: const EdgeInsets.only(bottom: 8),
                  child: Padding(
                    padding: const EdgeInsets.all(12),
                    child: Text(insight, style: theme.textTheme.bodyMedium),
                  ),
                ),
              ),
              const SizedBox(height: 24),
            ],
            _buildCategoryDistribution(context, distribution, statsProvider),
            const SizedBox(height: 24),
          ],
        ),
      ),
    );
  }

  Widget _buildProfileHeader(BuildContext context, User? user) {
    if (user == null) return const SizedBox();

    final semantic = Theme.of(context).extension<TimetySemanticColors>()!;
    final streak = user.currentStreak;
    final title = user.userTitle;
    final emoji = user.levelEmoji;
    final xpInLevel = user.xp;

    return Card(
      color: Theme.of(context).colorScheme.surfaceContainerLow,
      child: Padding(
        padding: const EdgeInsets.all(16),
        child: Column(
          children: [
            Row(
              children: [
                Expanded(
                  child: Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      Text(
                        user.name,
                        style: Theme.of(context).textTheme.headlineSmall,
                      ),
                      const SizedBox(height: 4),
                      Text(
                        '$emoji $title',
                        style: TextStyle(
                          color: semantic.focus,
                          fontSize: Theme.of(
                            context,
                          ).textTheme.bodyMedium?.fontSize,
                        ),
                      ),
                    ],
                  ),
                ),
                Icon(
                  Icons.local_fire_department,
                  color: semantic.warning,
                  size: 32,
                ),
                const SizedBox(width: 8),
                Text('$streak', style: Theme.of(context).textTheme.titleLarge),
              ],
            ),
            const SizedBox(height: 16),
            XPBar(currentXp: xpInLevel, maxXp: 2000, level: user.level),
          ],
        ),
      ),
    );
  }

  Widget _buildTodayProgress(
    BuildContext context,
    User? user,
    int todayFocusTime,
  ) {
    final semantic = Theme.of(context).extension<TimetySemanticColors>()!;
    final target = user?.dailyFocusTarget ?? 7200000;
    final progress = todayFocusTime / target;
    final hours = todayFocusTime ~/ 3600000;
    final minutes = (todayFocusTime % 3600000) ~/ 60000;

    return Column(
      children: [
        Text("Today's Progress", style: Theme.of(context).textTheme.titleLarge),
        const SizedBox(height: 16),
        GestureDetector(
          onTap: () => Navigator.push(
            context,
            MaterialPageRoute(
              builder: (_) => DailyStatsScreen(initialDate: DateTime.now()),
            ),
          ),
          child: RadialGraph(
            progress: progress.clamp(0.0, 1.0),
            text: '${hours}h ${minutes}m',
            size: 150,
            color: semantic.focus,
          ),
        ),
      ],
    );
  }

  Widget _buildWeeklyChart(BuildContext context, Map<String, int> weeklyData) {
    final semantic = Theme.of(context).extension<TimetySemanticColors>()!;
    final maxDuration = weeklyData.values.fold(
      1,
      (max, d) => d > max ? d : max,
    );

    return Card(
      color: Theme.of(context).colorScheme.surfaceContainerLow,
      child: Padding(
        padding: const EdgeInsets.all(16),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Text(
              "This Week's Focus",
              style: Theme.of(context).textTheme.titleMedium,
            ),
            const SizedBox(height: 16),
            ...weeklyData.entries.map((entry) {
              final barProgress = entry.value / maxDuration;
              return Padding(
                padding: const EdgeInsets.symmetric(vertical: 4),
                child: Column(
                  children: [
                    Row(
                      mainAxisAlignment: MainAxisAlignment.spaceBetween,
                      children: [
                        Text(
                          entry.key,
                          style: Theme.of(context).textTheme.labelMedium,
                        ),
                        Text(
                          '${entry.value ~/ 60000} min',
                          style: Theme.of(context).textTheme.labelSmall,
                        ),
                      ],
                    ),
                    const SizedBox(height: 4),
                    LinearProgressIndicator(
                      value: barProgress.clamp(0.0, 1.0),
                      minHeight: 8,
                      borderRadius: BorderRadius.circular(4),
                      color: semantic.focus,
                    ),
                  ],
                ),
              );
            }),
          ],
        ),
      ),
    );
  }

  Widget _buildCategoryDistribution(
    BuildContext context,
    Map<int, int> distribution,
    StatsProvider provider,
  ) {
    if (distribution.isEmpty) return const SizedBox();
    final total = distribution.values.fold(0, (sum, d) => sum + d);

    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Text(
          "Category Distribution",
          style: Theme.of(context).textTheme.titleLarge,
        ),
        const SizedBox(height: 16),
        ...distribution.entries.map((entry) {
          final category = provider.categories.firstWhere(
            (c) => c.id == entry.key,
            orElse: () =>
                Category(name: 'Unknown', colorHex: '#808080', iconName: ''),
          );
          final duration = entry.value;
          final catHours = duration ~/ 3600000;
          final catMinutes = (duration % 3600000) ~/ 60000;

          return Column(
            children: [
              Row(
                mainAxisAlignment: MainAxisAlignment.spaceBetween,
                children: [
                  Text(category.name),
                  Text('${catHours}h ${catMinutes}m'),
                ],
              ),
              const SizedBox(height: 4),
              LinearProgressIndicator(
                value: (duration / total).clamp(0.0, 1.0),
                color: Color(
                  int.parse(category.colorHex.replaceAll('#', '0xFF')),
                ),
                backgroundColor: Theme.of(
                  context,
                ).colorScheme.surfaceContainerHighest,
              ),
              const SizedBox(height: 12),
            ],
          );
        }),
      ],
    );
  }
}
